libtesting-ui/src/main/java/com/mapbox/navigation/testing/ui/BaseTest.kt

```diff
@@ -0,0 +1,51 @@
+package com.mapbox.navigation.testing.ui
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import androidx.test.uiautomator.UiDevice
+import com.mapbox.navigation.testing.ui.http.MockWebServerRule
+import com.schibsted.spain.barista.rule.cleardata.ClearDatabaseRule
+import com.schibsted.spain.barista.rule.cleardata.ClearFilesRule
+import com.schibsted.spain.barista.rule.cleardata.ClearPreferencesRule
+import org.junit.Rule
+
+open class BaseTest<A : AppCompatActivity>(activityClass: Class<A>) {
+
+    @get:Rule
+    val activityRule = ActivityTestRule(activityClass)
+
+    @get:Rule
+    val mockLocationUpdatesRule = MockLocationUpdatesRule()
+
+    @get:Rule
+    val mockWebServerRule = MockWebServerRule()
+
+    // Clear all app's SharedPreferences
+    @get:Rule
+    val clearPreferencesRule = ClearPreferencesRule()
+
+    // Delete all tables from all the app's SQLite Databases
+    @get:Rule
+    val clearDatabaseRule = ClearDatabaseRule()
+
+    // Delete all files in getFilesDir() and getCacheDir()
```

Comment:
Is this one to clear the files for storing results from the tests?

Reply:
This is to clear all of the cached resources generated by the test so that all tests run in full isolation.